tiatoolbox/models/architecture/unet.py

```diff
@@ -293,6 +312,8 @@ def create_block(kernels, input_ch, output_ch):
             next_up_ch = ch
             if ch_idx + 2 < len(down_ch_list):
                 next_up_ch = down_ch_list[ch_idx + 2]
+            if self.skip_type == "concat":
+                ch *= 2
```

Comment:
Ill condition, if you decide to allow arbitray `encoder_levels`, the number of channels are no longer guarantee to be a multiplication of 2 but of arbitrary number. A dummy forward within is a must to retrieve level and set the decoder input.

Reply:
As we discussed before,  the number of channel don't need to be multiplication of 2 but they will be multiplied by 2. With the genius design of encoder-decoder parts we can make sure that number of feature map from skip connection are the same as the number feature maps from upsampling part in the decoder. I have checked the code and resulted network architecture, this works with both custom number of layers and feature maps.